opm/simulators/flow/LogOutputHelper.cpp

```diff
@@ -557,11 +557,9 @@ production(const std::size_t reportStepNum,
 
         if (values[Ix::WaterRate] == 0.0) {
             values[Ix::WatGasRatio] = 0;
-        } else {
+        } else if (values[Ix::GasRate]) {
```

Comment:
Please make this more explicit (like below `values[Ix::GasRate] != 0)  to limit the guess work.